inventory-generation/identity-management/main.yml

```diff
@@ -30,27 +34,32 @@
         - "ocp-admin-credentials.json"
         - "ocp-ldap-sa-credentials.json"
 
-    - name: "Add users to inventory"
-      set_fact:
-        users: "{{ (users | default([])) + [ { 'first_name': (item.first_name | trim), 'last_name': (item.last_name | trim), 'email': (item.email | trim) , 'user_name': (item.email.split('@')[0] | trim) } ] }}"
+    - include_tasks: process_user.yml
+      loop_control:
+        loop_var: user_item
       loop: "{{ engagement_users }}"
 
-    - name: "Add LDAP Service Account"
+    - include_tasks: process_groups.yml
+      loop_control:
+        loop_var: group_item
+      loop: "{{ user_options.user_roles.options }}"
+
+    - name: "Assemble final list of users"
       set_fact:
-        users: "{{ (users | default([])) + [ { 'first_name': 'LDAP', 'last_name': 'SA', 'email': 'ldap@localhost.com', 'user_name': ocp_ldap_sa_username, 'generate_password': False, 'notify_user': False, 'password': ocp_ldap_sa_password  } ] }}"
+        users: "{{ user_list.keys() | map('extract', user_list) | list | flatten | unique }}"
+      when: users is defined
```

Comment:
`when: users is defined` will never be true from what I can tell - did you mean to check for `user_list` here?

Reply:
Yes, it should be `user_list` and I am wander how it worked for `users` variable. Fixed in be7fc14a1cf9ee1db5732eba8f1e21e79d7c2760